src/netex_idf/calendars.rs

```diff
@@ -0,0 +1,522 @@
+                DayTypeAssignment::InactiveDay(date) => is_included && *date != self.current_date,
+            });
+            if is_included {
+                return Some(self.current_date);
+            }
+        }
+        self.next()
+    }
+}
+
+fn parse_validity_period(valid_between: &Element) -> Result<ValidityPeriod> {
+    fn parse_date(valid_between: &Element, node: &str) -> Result<Date> {
+        let date = valid_between
+            .try_only_child(node)?
+            .text()
+            .parse::<NaiveDateTime>()?
+            .date();
+        Ok(date)
+    }
+    let start_date = parse_date(valid_between, "FromDate")?;
+    let end_date = parse_date(valid_between, "ToDate")?;
+    Ok(ValidityPeriod {
+        start_date,
+        end_date,
+    })
+}
+
+fn parse_operating_periods<'a, I>(operating_period_elements: I) -> OperatingPeriods
+where
+    I: Iterator<Item = &'a Element>,
+{
+    operating_period_elements
+        .filter_map(|operating_period_element| {
+            let id = operating_period_element.attribute("id")?;
+            let validity_period = parse_validity_period(operating_period_element).ok()?;
+            Some((id, validity_period))
+        })
+        .collect()
+}
+
+fn parse_day_type_assignments<'a, I>(
+    day_type_assignment_elements: I,
+    operating_periods: &OperatingPeriods,
+) -> DayTypeAssignments
+where
+    I: Iterator<Item = &'a Element>,
+{
+    let mut day_type_assignments = DayTypeAssignments::default();
+    for dta_element in day_type_assignment_elements {
+        let day_type_ref: String = skip_fail!(dta_element
+            .try_only_child("DayTypeRef")
+            .and_then(|dtr_element| dtr_element.try_attribute("ref")));
+        let operating_period_result = dta_element
+            .try_only_child("OperatingPeriodRef")
+            .and_then(|opr_element| opr_element.try_attribute::<String>("ref"))
+            .and_then(|op_ref| {
+                operating_periods.get(&op_ref).ok_or_else(|| {
+                    format_err!(
+                        "DayTypeAssignment '{}' references OperatingPeriod '{}' which doesn't exist",
+                        dta_element.attribute("id").unwrap_or_else(||String::new()),
+                        op_ref
+                    )
+                })
+            });
+        let date_result = dta_element.try_only_child("Date").and_then(|date_element| {
+            date_element
+                .text()
+                .parse::<Date>()
+                .map_err(|e| format_err!("{}", e))
+        });
+        let is_available_result =
+            dta_element
+                .try_only_child("isAvailable")
+                .and_then(|is_available_element| {
+                    is_available_element
+                        .text()
+                        .parse::<bool>()
+                        .map_err(|e| format_err!("{}", e))
+                });
```

Comment:
I actually refactored the whole function because of your comment. I think the new version is more accurate to what we expect (test for `Option` of the elements, and if we have them, log errors and skip the element if we can't find what we need). See 4ae4a676b48c85fbf0a2de539b730f19df4f7070.